src/modules/referral/referralService.ts

```diff
@@ -0,0 +1,30 @@
+import fetch, { AuthIdentity } from 'decentraland-crypto-fetch'
+import { handleErrorWithContext } from '../../shared/utils/errorHandler'
+import { config } from '../config'
+
+const REFERRAL_SERVER_URL = config.get('REFERRAL_SERVER_URL')
+
+const trackReferralProgress = async (referrer: string, identity: AuthIdentity, method: 'POST' | 'PATCH' = 'POST', account?: string) => {
```

Comment:
If you want to, you can create a hook that exposes the function `track` or `trackReferral` to track the referral progress. You can get from a hook the current identity.